File: Gemini/4. Desarrollo Web Endpoint RESTful/Javascript/endpoint.js
const express = require('express');
const app = express();
const port = 3000;

app.get('/saludo', (req, res) => {
  const nombre = req.query.nombre;
  if (nombre) {
    res.json({ mensaje: `Hola, ${nombre}` });
  } else {
    res.status(400).json({ error: 'El parámetro "nombre" es requerido' });
  }
});

app.listen(port, () => {
  console.log(`Servidor escuchando en http://localhost:${port}`);
});

File: Gemini/4. Desarrollo Web Endpoint RESTful/Ruby/endpoint.rb
require 'sinatra'
require 'json'

get '/saludo' do
  nombre = params['nombre']
  if nombre
    content_type :json
    { mensaje: "Hola, #{nombre}" }.to_json
  else
    status 400
    content_type :json
    { error: 'El parámetro "nombre" es requerido' }.to_json
  end
end